src/main/java/com/facebook/openwifirrm/ucentral/UCentralUtils.java

```diff
@@ -79,10 +83,45 @@ public static List<WifiScanEntry> parseWifiScanEntries(
 			for (JsonElement e : scanInfo) {
 				WifiScanEntry entry = gson.fromJson(e, WifiScanEntry.class);
 				entry.unixTimeMs = timestampMs;
+				// extract relevant IEs
```

Comment:
can you move this into a separate method? (this function is getting too long, and for example we may want to unit test IE parsing only)